test/FS.Embedded.Tests/Manifest/ManifestEntryTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    public class ManifestEntryTests
+    {
+        [Fact]
+        public void TraversingAFile_ReturnsUnknownPath()
+        {
+            // Arrange
+            var file = new ManifestFile("a", "a.b.c");
+
+            // Act
+            var result = file.Traverse(".");
+
+            // Assert
+            Assert.Equal(ManifestEntry.UnknownPath, result);
+        }
+
+        [Fact]
+        public void TraversingANonExistingFile_ReturnsUnknownPath()
+        {
+            // Arrange
+            var directory = ManifestDirectory.CreateDirectory("a", Array.Empty<ManifestEntry>());
+
+            // Act
+            var result = directory.Traverse("missing.txt");
+
+            // Assert
+            Assert.Equal(ManifestEntry.UnknownPath, result);
+        }
+
+        [Fact]
+        public void TraversingWithDot_ReturnsSelf()
+        {
+            // Arrange
+            var directory = ManifestDirectory.CreateDirectory("a", Array.Empty<ManifestEntry>());
+
+            // Act
+            var result = directory.Traverse(".");
+
+            // Assert
+            Assert.Same(directory, result);
+        }
+
+        [Fact]
+        public void TraversingWithDotDot_ReturnsParent()
+        {
+            // Arrange
+            var childDirectory = ManifestDirectory.CreateDirectory("b", Array.Empty<ManifestEntry>());
+            var directory = ManifestDirectory.CreateDirectory("a", new[] { childDirectory });
+
+            // Act
+            var result = childDirectory.Traverse("..");
+
+            // Assert
+            Assert.Equal(directory, result);
+        }
+
+        [Fact]
+        public void TraversingRootDirectoryWithDotDot_ReturnsSinkDirectory()
+        {
+            // Arrange
+            var directory = ManifestDirectory.CreateRootDirectory(Array.Empty<ManifestEntry>());
+
+            // Act
+            var result = directory.Traverse("..");
```

Comment:
One more test for `..\..`, plus ones where we `CreateRootDirectory()` and then attempt to navigate past root.

Reply:
Those tests are covered by TraversingUnknownPath_ReturnsSinkDirectory unless i'm not understanding something

Reply:
More along the lines of 

```
var directory = ManifestDirectory.CreateRootDirectory(
  ManifestDirectory.CreateDirectory("a", "test1.txt"),
  ManifestDirectory.CreateDirectory("b", "test2.txt"));

var newRoot = directory.Traverse("a").ToRootDirectory();
Assert.Same(UnknownPath, newRoot.Traverse("../b/test.txt"));
```